HolidayList/Holiday.cs

```diff
@@ -1,11 +1,404 @@
+//                case "Worldwide observance":
+//                    return TypeElement.WorldwideObservance;
+//            }
+//            throw new Exception("Cannot unmarshal type TypeElement");
+//        }
+
+//        public override void WriteJson(JsonWriter writer, object untypedValue, JsonSerializer serializer)
+//        {
+//            if (untypedValue == null)
+//            {
+//                serializer.Serialize(writer, null);
+//                return;
+//            }
+//            var value = (TypeElement)untypedValue;
+//            switch (value)
+//            {
+//                case TypeElement.Christian:
+//                    serializer.Serialize(writer, "Christian");
+//                    return;
+//                case TypeElement.ClockChangeDaylightSavingTime:
+//                    serializer.Serialize(writer, "Clock change/Daylight Saving Time");
+//                    return;
+//                case TypeElement.Hebrew:
+//                    serializer.Serialize(writer, "Hebrew");
+//                    return;
+//                case TypeElement.Hinduism:
+//                    serializer.Serialize(writer, "Hinduism");
+//                    return;
+//                case TypeElement.LocalHoliday:
+//                    serializer.Serialize(writer, "Local holiday");
+//                    return;
+//                case TypeElement.LocalObservance:
+//                    serializer.Serialize(writer, "Local observance");
+//                    return;
+//                case TypeElement.Muslim:
+//                    serializer.Serialize(writer, "Muslim");
+//                    return;
+//                case TypeElement.NationalHoliday:
+//                    serializer.Serialize(writer, "National holiday");
+//                    return;
+//                case TypeElement.Observance:
+//                    serializer.Serialize(writer, "Observance");
+//                    return;
+//                case TypeElement.Orthodox:
+//                    serializer.Serialize(writer, "Orthodox");
+//                    return;
+//                case TypeElement.Season:
+//                    serializer.Serialize(writer, "Season");
+//                    return;
+//                case TypeElement.SportingEvent:
+//                    serializer.Serialize(writer, "Sporting event");
+//                    return;
+//                case TypeElement.UnitedNationsObservance:
+//                    serializer.Serialize(writer, "United Nations observance");
+//                    return;
+//                case TypeElement.WorldwideObservance:
+//                    serializer.Serialize(writer, "Worldwide observance");
+//                    return;
+//            }
+//            throw new Exception("Cannot marshal type TypeElement");
+//        }
+
+//        public static readonly TypeElementConverter Singleton = new TypeElementConverter();
+//    }
+//}
+
+
+
+// CODE REVIEW - UMESH SINGH
```

Comment:
Put discussion like this in the GitHub pull request.  In code comments, it's clutter.